source/include/FreeRTOS_Sockets.h

```diff
@@ -179,12 +179,12 @@
         uint32_t sin_flowinfo;    /**< IPv6 flow information, not used in this library. */
         IP_Address_t sin_address; /**< The IPv4/IPv6 address. */
     };
-    #define sin_addr              sin_address.ulIP_IPv4
-    #define sin_addr4             sin_address.ulIP_IPv4
-    #define sin_addr6             sin_address.xIP_IPv6
-    #define sin_addrv4            sin_address.ulIP_IPv4
-    #define sin_addrv6            sin_address.xIP_IPv6
-    #define freertos_sockaddr6    freertos_sockaddr
```

Comment:
For me this is a bit painful, because the old code was a trick to get older IPv6/multi programs compiled without a pain.

But I agree, the set of macros can lead to confusion. It is better to show the nested structs:

~~~diff
-    pxAddress->sin_addr
+    pxAddress->sin_address.ulIP_IPv4
~~~
Also when debugging, the IDE won't understand a macro like `sin_addr`, and it will show the value of a real field like `ulIP_IPv4`.